### DISCUSSION TITLE
Blade Attributes modifiers

### DISCUSSION THREAD

**Discussion by ph7jack** [laravel/framework#40659] [Ideas]
I'm creating a component, but i need the attributes modifiers list, ex:
`<x-button tooltip.left.top="Click To Save">Click</x-button>`

I need to get the left and top modifiers.
When check if the tooltip attribute exists, its return false `$attributes->has('tooltip')`
Is false because the `tooltip.left.top` is set together. 
I can implement this feature, but I don know if will be accepted.

I think:

```
                     return an object        return array or collect
$attributes      ->get('tooltip')                ->modifiers()
```

or

$attributes->getModifiers(): Collection or array

@taylorotwell

**Comment by aguingand** [comment]
You can write a macro for ComponentAttributesBag, look at livewire implementation:
- https://github.com/livewire/livewire/blob/0d417b27791af09c79108eafd1344842f83a26ee/src/LivewireServiceProvider.php#L245
- https://github.com/livewire/livewire/blob/0d417b27791af09c79108eafd1344842f83a26ee/src/WireDirective.php

If you name the macro "directive", you could do `$attributes->directive('tooltip')->modifiers()`

**Reply by aguingand** [reply]
I think the modifiers idea is too specific to be added in the framework
